### PR TITLE
chore(remix-server-runtime): standardize on uppercase method

### LINKED ISSUE #145: Normalize fetch methods to uppercase
Apparently `PATCH` is case-sensitive, so we should just normalize all HTTP methods to uppercase when submitting forms via `fetch`.

### PR DESCRIPTION
**Pull request by donavon**
According to the [Fetch spec](https://fetch.spec.whatwg.org/#methods), method names are case insensitive _except_ for standard names `DELETE`, `GET`, `HEAD`, `OPTIONS`, `POST`, and `PUT` which are all normalized to "byte uppercase".

The only exception is `PATCH` which was never made the spec. For `PATCH` you will need to upper case it before you compare.

The PR brings consistency with the spec as well as saves the need to `toLowerCase` on methods that are guaranteed to be upper case.

---

Closes #145

### CODE REVIEW

File: packages/remix-server-runtime/server.ts [hunk 1]
```diff
@@ -544,24 +544,19 @@ async function handleResourceRequest({
...
   return (
-    method === "post" ||
-    method === "put" ||
-    method === "patch" ||
-    method === "delete"
+    method === "POST" ||
+    method === "PUT" ||
+    method === "DELETE" ||
+    method.toUpperCase() === "PATCH"
   );
```
Comment (MichaelDeBoey) [inline packages/remix-server-runtime/server.ts:553]: Maybe we could even simplify this to

```suggestion
    return ["POST", "PUT", "DELETE"].includes(method.toUpperCase());
```

**Comment by kiliman** [comment]
I was curious it this would break if a client sent lowercase methods, but apparently `fetch` and `node-fetch` both covert method to uppercase before generating the request. So by the time it reaches Remix, the method should be normalized to uppercase.

File: packages/remix-server-runtime/server.ts (hunk 1 above)
Comment (donavon) [reply packages/remix-server-runtime/server.ts]: see new commit. no need to create/destroy the array each time

**Comment by donavon** [comment]
> the method should be normalized to uppercase.

per the Fetch spec, yes

**Comment by donavon** [comment]
Now using `Set`, which is perfect. In database terminology, execution is "indexed" with `Set` vs a "full table scan" with `includes`

**Review by MichaelDeBoey — APPROVED** [review]

**Review by sergiodxa — APPROVED** [review]

**Comment by machour** [comment]
Should fix https://github.com/remix-run/remix/issues/145

**Review by machour — APPROVED** [review]

**Pull request merged**